File: frontend/src/types/health.ts
export type EngineeringHealthStatus =
  | "excellent"
  | "healthy"
  | "needs_attention"
  | "at_risk";

export type HealthImpact =
  | "positive"
  | "negative"
  | "neutral";

export interface HealthEvidence {
  label: string;
  value: string;
  impact: HealthImpact;
}

export interface HealthComponent {
  score: number;
  weight: number;
  weighted_score: number;
  calculation_basis: string[];
}

export interface EngineeringHealth {
  project_id: number;
  score: number;
  status: EngineeringHealthStatus;
  generated_at: string;
  lookback_days: number;

  issue_health: HealthComponent;
  cicd_reliability: HealthComponent;
  delivery_activity: HealthComponent;
  github_activity: HealthComponent;

  evidence: HealthEvidence[];
}

File: backend/app/schemas/health.py
from datetime import datetime

from pydantic import BaseModel


class IssueMetrics(BaseModel):
    total: int
    open: int
    todo: int
    in_progress: int
    in_review: int
    done: int
    critical: int
    high_priority: int
    unassigned: int


class ProjectHealth(BaseModel):
    project_id: int
    health: str
    health_score: int
    issues: IssueMetrics
class HealthComponent(BaseModel):
    score: float
    weight: float
    weighted_score: float
    calculation_basis: list[str]


class HealthEvidence(BaseModel):
    label: str
    value: str
    impact: str


class EngineeringHealth(BaseModel):
    project_id: int
    score: float
    status: str
    generated_at: datetime
    lookback_days: int

    issue_health: HealthComponent
    cicd_reliability: HealthComponent
    delivery_activity: HealthComponent
    github_activity: HealthComponent

    evidence: list[HealthEvidence]